wrangler-service/src/main/java/co/cask/wrangler/service/connections/ConnectionHandler.java

```diff
@@ -140,29 +140,27 @@ private void validateAndCreateDefaultConnections() {
    * @param responder Responder for data going out.
    */
   @POST
-  @Path("connections/create")
-  public void create(HttpServiceRequest request, HttpServiceResponder responder) {
-    // Extract the body of the request and transform it to the Connection object.
-    RequestExtractor extractor = new RequestExtractor(request);
-    ConnectionMeta connection;
-    try {
-      connection = extractor.getConnectionMeta();
-    } catch (IllegalArgumentException e) {
-      ServiceUtils.error(responder, HttpURLConnection.HTTP_BAD_REQUEST, e.getMessage());
-      return;
-    }
+  @Path("contexts/{context}/connections/create")
+  public void create(HttpServiceRequest request, HttpServiceResponder responder,
+                     @PathParam("context") String namespace) {
+    respond(request, responder, namespace, () -> {
+      // Extract the body of the request and transform it to the Connection object.
+      RequestExtractor extractor = new RequestExtractor(request);
+      ConnectionMeta connection = extractor.getConnectionMeta();
 
-    try {
       // Create an instance of the connection, if the connection id already exists,
       // it will throw an exception.
       String id = store.create(connection);
       // Return the id in the response.
-      responder.sendJson(new ServiceResponse<>(ImmutableList.of(id)));
-    } catch (ConnectionAlreadyExistsException e) {
-      ServiceUtils.error(responder, HttpURLConnection.HTTP_CONFLICT, e.getMessage());
-    } catch (Exception e) {
-      ServiceUtils.error(responder, e.getMessage());
-    }
+      return new ServiceResponse<>(id);
```

Comment:
should this be `new ServiceResponse<>(ImmutableList.of(id)`?

Reply:
Oh I incorrectly thought this was a convenience method for a single value, but you are right this would interpret the id as a message. Will fix. I'm not sure how the UI was working....